docs/apps.md

```diff
@@ -0,0 +1,505 @@
+      "slug": "emails",
+      "icon": "/Apps/marketplace/emails.svg",
+      "source": "git://github.com/cozy/cozy-emails",
+      "default_locale": "en",
+      "description": "A webmail for Cozy Cloud",
+      "locales": {
+        "fr": {
+          "name": "courriels",
+          "description": "Un client web pour les courriels dans Cozy Cloud"
+        }
+      }
+    }
+  }
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 201 Created
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "data": {
+    "id": "4f6436ce-8967-11e6-b174-ab83adac69f2",
+    "type": "io.cozy.manifests",
+    "attributes": {
+      "name": "cozy-emails",
+      "slug": "emails",
+      "icon": "/Apps/marketplace/emails.svg",
+      "source": "git://github.com/cozy/cozy-emails",
+      "default_locale": "en",
+      "description": "A webmail for Cozy Cloud",
+      "locales": {
+        "fr": {
+          "name": "courriels",
+          "description": "Un client web pour les courriels dans Cozy Cloud"
+        }
+      }
+    }
+  }
+}
+```
+
+### PATCH /apps/manifests/:id
+
+Update an application in the marketplace.
+
+### DELETE /apps/manifests/:id
+
+Remove an application from the marketplace.
+
+
+Uninstall an application
+------------------------
+
+### DELETE /apps/:slug
+
+#### Request
+
+```http
+DELETE /apps/tasky HTTP/1.1
+```
+
+#### Response
+
+```http
+HTTP/1.1 204 No Content
+```
+
+
+Access an application
+---------------------
+
+Each application will run on its sub-domain. The sub-domain is the slug used
+when installing the application (`calendar.cozy.example.org` if it was
+installed via a POST on /apps/calendar). On the main domain
+(`cozy.example.org`), there will be the registration process, the login form,
+and it will redirect to `home.cozy.example.org` for logged-in users.
+
+### Rationale
+
+The applications have different roles and permissions. An application is
+identified when talking to the stack via a token. The token has to be injected
+in the application, one way or another, and it have to be unaccessible from
+other apps.
+
+If the applications run on the same domain (for example
+`https://cozy.example.org/apps/calendar`), it's nearly impossible to protect
+an application to take the token of another application. The security model of
+the web is based too heavily on [Same Origin
+Policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy)
+for that. If the token is put in the html of the index.html page of an app,
+another app can use the fetch API to get the html page, parse its content and
+extract the token. We can think of other ways to inject the token (indexeddb,
+URL, cookies) but none will provide a better isolation. Maybe in a couple
+years, when [the origin spec](https://w3c.github.io/webappsec-suborigins/)
+will be more advanced.
+
+So, if having the apps on the same domain is not possible, we have to put them
+on several domains. One interesting way to do that is using sandboxed iframes.
```

Comment:
Done. It's difficult for me to explain clearly the issues here. The advices are welcomed :smile: